internal/cutil/iovec.go

```diff
@@ -2,77 +2,60 @@ package cutil
 
 /*
 #include <stdlib.h>
+#include <string.h>
 #include <sys/uio.h>
 */
 import "C"
-
 import (
 	"unsafe"
 )
 
-var iovecSize uintptr
-
-// StructIovecPtr is an unsafe pointer wrapping C's `*struct iovec`.
-type StructIovecPtr unsafe.Pointer
-
-// Iovec helps manage struct iovec arrays needed by some C functions.
+// Iovec is a slice of iovec structs. Might have allocated C memory, so it must
```

Comment:
"slice of iovec structs" is "leaking" implementation details. I'd just go with reminding the user that to use the type it should be Free()ed when no longer needed, and not mention what kinds of types may be used on the inside.